File: WarehouseManagementSystem/wwwroot/js/models/ActivityLog.js
class ActivityLog {
    constructor(data) {
        this.id = data.id || 0;
        this.actionType = data.actionType || 'Info';
        this.description = data.description || '';
        this.itemSKU = data.itemSKU || null;
        this.timestamp = data.timestamp ? new Date(data.timestamp) : new Date();
        this.userName = data.userName || 'System';
    }

    /**
     * Get the background color class based on action type
     * @returns {string} CSS class for background color
     */
    getBgColorClass() {
        switch (this.actionType) {
            case 'Add':
                return 'bg-blue-100';
            case 'Remove':
                return 'bg-red-100';
            case 'Update':
                return 'bg-yellow-100';
            case 'Move':
                return 'bg-green-100';
            case 'Error':
                return 'bg-pink-100';
            default:
                return 'bg-gray-100';
        }
    }

    /**
     * Get the text color class based on action type
     * @returns {string} CSS class for text color
     */
    getTextColorClass() {
        switch (this.actionType) {
            case 'Add':
                return 'text-blue-600';
            case 'Remove':
                return 'text-red-600';
            case 'Update':
                return 'text-yellow-600';
            case 'Move':
                return 'text-green-600';
            case 'Error':
                return 'text-pink-600';
            default:
                return 'text-gray-600';
        }
    }

    /**
     * Get the icon class based on action type
     * @returns {string} CSS class for icon
     */
    getIconClass() {
        switch (this.actionType) {
            case 'Add':
                return 'ri-add-line';
            case 'Remove':
                return 'ri-subtract-line';
            case 'Update':
                return 'ri-edit-line';
            case 'Move':
                return 'ri-arrow-left-right-line';
            case 'Error':
                return 'ri-error-warning-line';
            default:
                return 'ri-information-line';
        }
    }

    /**
     * Get a friendly time display (e.g., "2 hours ago")
     * @returns {string} Friendly time display
     */
    getFriendlyTimeDisplay() {
        const now = new Date();
        const diff = now - this.timestamp;
        
        // Convert milliseconds to seconds
        const seconds = Math.floor(diff / 1000);
        
        if (seconds < 60) {
            return 'Just now';
        }
        
        // Convert seconds to minutes
        const minutes = Math.floor(seconds / 60);
        
        if (minutes < 60) {
            return `${minutes} minute${minutes !== 1 ? 's' : ''} ago`;
        }
        
        // Convert minutes to hours
        const hours = Math.floor(minutes / 60);
        
        if (hours < 24) {
            return `${hours} hour${hours !== 1 ? 's' : ''} ago`;
        }
        
        // Convert hours to days
        const days = Math.floor(hours / 24);
        
        if (days < 7) {
            return days === 1 ? 'Yesterday' : `${days} days ago`;
        }
        
        // Format date for anything older than a week
        return this.timestamp.toLocaleDateString('en-US', {
            year: 'numeric',
            month: 'short',
            day: 'numeric'
        });
    }
}

// Make class available globally
window.ActivityLog = ActivityLog; 

File: WarehouseManagementSystem/Models/ActivityLog.cs
using System;
using System.ComponentModel.DataAnnotations;
using WarehouseManagementSystem.Services.Interfaces;

namespace WarehouseManagementSystem.Models
{
    public class ActivityLog : IActivity
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public required string ActionType { get; set; } // "Add", "Remove", "Update", "Move", "Login"

        [Required]
        public required string Description { get; set; }

        public string? ItemSKU { get; set; }

        public string? UserId { get; set; }

        public string? UserName { get; set; }

        public DateTime Timestamp { get; set; } = DateTime.Now;

        // IActivity implementation
        public string GetTitle()
        {
            return Description.Split(' ')[0]; // Get the first word as a short title
        }

        public string GetDescription()
        {
            return Description;
        }

        public string GetIcon()
        {
            return GetIconClass();
        }

        public string GetTimestamp()
        {
            // Format relative timestamp
            var now = DateTime.Now;
            var diff = now - Timestamp;

            if (diff.TotalSeconds < 60)
                return "Just now";

            if (diff.TotalMinutes < 60)
                return $"{(int)diff.TotalMinutes} minute{(diff.TotalMinutes >= 2 ? "s" : "")} ago";

            if (diff.TotalHours < 24)
                return $"{(int)diff.TotalHours} hour{(diff.TotalHours >= 2 ? "s" : "")} ago";

            if (diff.TotalDays < 7)
                return $"{(int)diff.TotalDays} day{(diff.TotalDays >= 2 ? "s" : "")} ago";

            if (diff.TotalDays < 30)
            {
                int weeks = (int)(diff.TotalDays / 7);
                return $"{weeks} week{(weeks >= 2 ? "s" : "")} ago";
            }

            return Timestamp.ToString("MMM d, yyyy");
        }

        public string GetColorClasses()
        {
            return $"{GetBgColorClass()} {GetTextColorClass()}";
        }

        public string GetItemLink()
        {
            return !string.IsNullOrEmpty(ItemSKU) ? $"/Items/Details/{ItemSKU}" : "#";
        }

        // Helper method to get appropriate icon class
        public string GetIconClass()
        {
            return ActionType.ToLower() switch
            {
                "add" => "ri-add-line",
                "remove" => "ri-subtract-line",
                "update" => "ri-edit-line",
                "move" => "ri-arrow-left-right-line",
                "login" => "ri-login-box-line",
                "logout" => "ri-logout-box-line",
                _ => "ri-information-line"
            };
        }

        // Helper method to get background color class
        public string GetBgColorClass()
        {
            return ActionType.ToLower() switch
            {
                "add" => "bg-blue-100",
                "remove" => "bg-red-100",
                "update" => "bg-yellow-100",
                "move" => "bg-green-100",
                "login" => "bg-purple-100",
                "logout" => "bg-gray-100",
                _ => "bg-gray-100"
            };
        }

        // Helper method to get text color class
        public string GetTextColorClass()
        {
            return ActionType.ToLower() switch
            {
                "add" => "text-blue-600",
                "remove" => "text-red-600",
                "update" => "text-yellow-600",
                "move" => "text-green-600",
                "login" => "text-purple-600",
                "logout" => "text-gray-600",
                _ => "text-gray-600"
            };
        }
    }
} 